modules/services/eternal-terminal.nix

```diff
@@ -0,0 +1,92 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let cfg = config.services.eternal-terminal;
+in {
+  options = {
+    services.eternal-terminal = {
+
+      enable = mkEnableOption (lib.mdDoc "Eternal Terminal server");
+
+      package = mkOption {
+        type = types.path;
+        default = pkgs.eternal-terminal;
+        defaultText = "pkgs.eternal-terminal";
+        description = lib.mdDoc
+          "This option specifies the eternal-terminal package to use.";
+      };
+
+      port = mkOption {
+        default = 2022;
+        type = types.port;
+        description = lib.mdDoc ''
+          The port the server should listen on. Will use the server's default (2022) if not specified.
+
+          Make sure to open this port in the firewall if necessary.
+        '';
+      };
+
+      verbosity = mkOption {
+        default = 0;
+        type = types.enum (lib.range 0 9);
+        description = lib.mdDoc ''
+          The verbosity level (0-9).
+        '';
+      };
+
+      silent = mkOption {
+        default = false;
+        type = types.bool;
+        description = lib.mdDoc ''
+          If enabled, disables all logging.
+        '';
+      };
+
+      logSize = mkOption {
+        default = 20971520;
+        type = types.int;
+        description = lib.mdDoc ''
+          The maximum log size.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    # We need to ensure the et package is fully installed because
+    # the (remote) et client runs the `etterminal` binary when it
+    # connects.
+    environment.systemPackages = [ cfg.package ];
+
+    launchd.daemons.eternal-terminal = {
+      path = [ cfg.package ];
+      serviceConfig = {
+        ProgramArguments = [
+          "${cfg.package}/bin/etserver"
+          "--daemon"
```

Comment:
It looks like `--daemon` causes the program to fork off, going by the `Type = "forking"` in the systemd services. I believe that launchd doesn't handle such programs properly (though my understanding may be out of date); is there any way to get the server to run in the foreground?

Reply:
I think you are right. I was not aware of that issue and I included `--daemon` in the arguments originally because that is what the [homebrew formula](https://github.com/MisterTea/homebrew-et/blob/master/et.rb#L25) does and I've been running eternal-terminal that way for a long time without noticing any problems. But, this prompted me to look around and I found that the project repository includes a [launchd config](https://github.com/MisterTea/EternalTerminal/blob/master/init/launchd/homebrew.mxcl.et.plist) that does not include `--daemon`. It was removed in this [commit](https://github.com/MisterTea/EternalTerminal/commit/1d9cd2be9dc1ec7694e9472004b7910bbb5c34cf) for reasons that sound like what you are referring to. Anyway, I updated the agent config here to align more with what's in the source repository.

Reply:
Thanks! The lack of `KeepAlive` is strange to me, but if it's in the upstream launchd daemon file then I assume there's probably some reason for it.

Reply:
I saw that you changed `KeepAlive` to `false` to match the upstream plist, however it looks like the reason why it is there in the first place is because they used to use `--daemonize` and launchd would lose track of the process and presumably keep launching it, so I think it makes sense to set it to `true`

Reply:
Ah, whoops, that makes perfect sense; merged before getting this comment. @ryane do you mind opening a PR to toggle `KeepAlive` back on?

Reply:
ah, that does make sense. here is the follow-up pr: #748